pandas-stubs/core/groupby/generic.pyi

```diff
@@ -110,18 +111,20 @@ class _DataFrameGroupByNonScalar(DataFrameGroupBy):
 class DataFrameGroupBy(GroupBy):
     def any(self, skipna: bool = ...) -> DataFrame: ...
     def all(self, skipna: bool = ...) -> DataFrame: ...
-    # mypy sees the two overloads as overlapping
+    # mypy and pyright see these overloads as overlapping
     @overload
     def apply(  # type: ignore[misc]
-        self, func: Callable[[DataFrame], Series | Scalar], *args, **kwargs
+        self, func: Callable[[Iterable], float], *args, **kwargs
```

Comment:
The `Callable[[Iterable]...` should come last, because you want the other ones to match first.  I think that will make `pyright` not see things as overlapping.

Reply:
I thought that originally. But then mypy gives the following error:
```
tests/test_frame.py:559: error: Incompatible types in assignment (expression has type "Series[Any]", variable has type "DataFrame")  [assignment]
```

I can't really explain that behaviour, because according to everything I can find mypy should also match in order.

Reply:
I just tried moving it last and it worked.  Still had `pyright` complain about the last overload never being matched without the `#type: ignore` in there, but it did work.  Can you try again?

Reply:
Apologies, looks like I had the previous version of mypy. Running the reordered code against the latest version works correctly.